presto-main/src/main/java/com/facebook/presto/operator/scalar/MathFunctions.java

```diff
@@ -327,6 +331,51 @@ public static long truncate(@SqlType(StandardTypes.REAL) long num)
         return floatToRawIntBits((float) (Math.signum(numInFloat) * Math.floor(Math.abs(numInFloat))));
     }
 
+    @Description("truncate to double by dropping digits after decimal point")
```

Comment:
If there's a limit for decimal point, I think we should be explicit about it and mention in description and in documentation. We should probably also fail the query if an out-of-range number is given, rather than silently changing the behavior.

Reply:
Though if the limit is actually limitation of double then this behavior is fine. But in that case I think we also don't have to special handle the max?

Reply:
Thanks for the review. Are you suggesting to document the max number of precision for double type in alignment with `MAX_DECIMAL_PLACES` value? Yes the limit only handles the double/real type behavior. If the max limit reaches/exceeded, the outcome should be either 0.0 or negative value. I have 2 tests covered that:

`assertFunction("truncate(DOUBLE '" + Double.MAX_VALUE + "', -408)", DOUBLE, 0.0);
assertFunction("truncate(DOUBLE '" + -Double.MAX_VALUE / 10 + "', 408)", DOUBLE, -Double.MAX_VALUE / 10);
`

Is this what you were thinking to cover? @rongrong

Reply:
I mean add this to the rst file. Maybe also add it to the function description. So people know the maximum value for decimals should be within 400. Also document the behavior if the provided decimal is larger than 400. Why do we choose 400 as the max digit though? What's the logic behind this?

Reply:
Actually I double-checked the oracle doc which indicates that the scale of the decimal points ranges from -84 to 127. I will change 400 to 127 to align with oracle's behavior and document it in rst file.